src/OpenTelemetry.ClrProfiler.Managed/Instrumentation.cs

```diff
@@ -65,6 +66,10 @@ public static void Initialize()
 
                     _tracerProvider = builder.Build();
                     Log("OpenTelemetry tracer initialized.");
+
+                    // Register to shutdown events
+                    AppDomain.CurrentDomain.ProcessExit += OnExit;
+                    AppDomain.CurrentDomain.DomainUnload += OnExit;
```

Comment:
.NET calls these events on a best effort basis. We might reduce the impact of the loss of exported spans, but it does not guarantee the that dispose will complete.

Do we know what is the recommendation from OpenTelemetry .NET SDK to handle such scenarios?

Reply:
Just more info about the issue >
https://stackoverflow.com/questions/20675024/when-will-appdomain-processexit-not-get-called

Not sure if something similar applies to IIS also, it seems additional layer of magic.

Reply:
We might not have another event which could help us dispose an object during shutdown. My earlier comment was to say there is no guarantee these events are triggered in all cases. But this approach should work with most of the applications.

Yes. This is applicable to IIS and have seen similar issue in the past.

Reply:
I believe that the situations that would terminate the app but not trigger these events would also likely cause not all data to be exported when the Simple processor is used instead of the batch processor. I think it's just a question of how likely is this problem to occur, and how much data would not get exported if that happens? In practice, I've also seen the time window that we have to send the data when the event is triggered, generally being long enough to send all of the data.

Reply:
Should we add "Data loss" section to the docs? 
Currently I understand these situations will cause a data loss:
- abnormal termination of the process
- unhandled exception terminates the process

Could reduce the impact (but not avoiding):
- use Simple processor
- use local OTel Collector for faster data handover

Reply:
I think is a good idea. Just instead of using simple processor I would suggest setting some BSP env var values

Reply:
> Do we know what is the recommendation from OpenTelemetry .NET SDK to handle such scenarios?

I'm not aware of any official guidance there. Most of the situations that the SDK is involved in, you are able to leverage a using block or statement to get the desired behavior, by initializing the SDK in an appropriate location in your code. This question is probably worth asking some of the SDK maintainers about.